### ISSUE TITLE
Can't replace existing webpack rules for css files

### ISSUE THREAD

**Issue by everdimension** [gatsbyjs/gatsby#16129]
### Description

Hi! I am trying to make gatsby work with all `.css` files as CSS Modules. I want to make it work without having to add a `.module.css` extension.

So far it doesn't seem to work.

### Steps to reproduce

I tried using the `onCreateWebpackConfig` hook to add a *css module* rule for all `\.css$` files.
But I have this error when building:

```
ERROR in ./src/assets/sizes.css (./node_modules/css-loader??ref--11-oneOf-1-1!./node_modules/gatsby-plugin-postcss/node_modules/postcss-loader/src??ref--11-oneOf-1-2!./node
_modules/style-loader??ref--16-0!./node_modules/css-loader??ref--16-1!./node_modules/postcss-loader/lib??postcss-3!./src/assets/sizes.css)
Module build failed (from ./node_modules/gatsby-plugin-postcss/node_modules/postcss-loader/src/index.js):
SyntaxError

(2:1) Unknown word

  1 |
> 2 | var content = require("!!../../node_modules/css-loader/index.js??ref--16-1!../../node_modules/postcss-loader/lib/index.js??postcss-3!./sizes.css");
```

What I found from googling is that this error can happen when there are multiple loaders trying to parse the same `.css` file. So I figured that this rule that I added probably conflicted with the default gatsby rule.
I have tried to remove the default rule by getting the config with the `getConfig()` method.
Code looks like this;

```js
exports.onCreateWebpackConfig = ({
  rules,
  actions,
  getConfig,
}) => {
  const config = getConfig();

  const cssRule = rules.css({ modules: true });
  cssRule.test = /\.css$/;
  delete cssRule.exclude;

  config.module.rules = [
    // exclude existing css rules
    ...config.module.rules.filter((rule) => {
      if (rule.test instanceof RegExp) {
        return !(rule.test.test('.css') || rule.test.test('.module.css'));
      }
      return true;
    }),
    // replace with custom css rule
    cssRule,
  ];

  actions.replaceWebpackConfig(config);
};
```

But the config that I get doesn't seem to have any rules for `\.css`.

### Expected result

I expect to be able to replace existing rules for `.css` and replace with them my own.

### Actual result

As far as I can tell the default rules aren't getting replaced.

### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here.

```
  System:
    OS: macOS 10.14.5
    CPU: (4) x64 Intel(R) Core(TM) i5-8210Y CPU @ 1.60GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 10.9.0 - ~/.nvm/versions/node/v10.9.0/bin/node
    Yarn: 1.15.2 - /usr/local/bin/yarn
    npm: 6.10.1 - ~/.nvm/versions/node/v10.9.0/bin/npm
  Languages:
    Python: 2.7.15 - /usr/local/bin/python
  Browsers:
    Chrome: 75.0.3770.142
    Firefox: 68.0.1
    Safari: 12.1.1
  npmPackages:
    gatsby: ^2.13.13 => 2.13.13
    gatsby-image: ^2.2.6 => 2.2.6
    gatsby-plugin-postcss: ^2.1.2 => 2.1.2
    gatsby-plugin-react-svg: ^2.1.1 => 2.1.1
    gatsby-plugin-root-import: ^2.0.5 => 2.0.5
    gatsby-plugin-sharp: ^2.2.8 => 2.2.8
    gatsby-plugin-typescript: ^2.1.2 => 2.1.2
    gatsby-source-filesystem: ^2.1.5 => 2.1.5
    gatsby-source-graphql: ^2.1.1 => 2.1.1
    gatsby-transformer-sharp: ^2.2.4 => 2.2.4
```

**Comment by everdimension** [comment]
Yeah I would still love to see this resolved :)

**Comment by universse** [comment]
Hi you still waiting on this? I might be able to take a look.

**Comment by everdimension** [comment]
@universse Hi! Yes. We're using a workaround now, but would love to see this issue fixed.

**Comment by universse** [comment]
```js
  const cssRule = {
    ...rules.cssModules(),
    test: rules.css().test
  }

  config.module.rules = [
    ...config.module.rules.filter(rule => {
      const areCssRules =
        rule.oneOf && rule.oneOf.some(r => r.test.test('style.css'))

      return !areCssRules
    }),
    cssRule
  ]
```

You can try this snippet. It seems to work with my local project. Lmk if something isn't working.

**Comment by everdimension** [comment]
Sorry for silence, I will try to confirm whether the solution works for us within a couple of days. Thanks!

**Comment by everdimension** [comment]
@universse Hey, I tried your solution and it worked!
Thanks!

Do you think this should be added to the docs? Because my initial approach was copied from the docs and didn't work, so it seems that this is a somewhat special case.

**Comment by universse** [comment]
I'm not too sure cuz this seems like an advanced use case. Can you link me to the docs you're referring to? I can't seem to find it.

**Comment by everdimension** [comment]
Yeah, I think it's here: https://github.com/gatsbyjs/gatsby/issues/16129
(it seems, though, that the instructions changed a bit since I last opened this page)

About your solution — I don't think it's obvious that `rules` have a `cssModules` method (`rules.cssModules()`), nor that you should search for existing style rules in the `oneOf` array.
Would be great to see this subtleties in the docs.

**Comment by universse** [comment]
I was experimenting with webpack config at the time I picked up on this issue. I managed to answer your issue by searching through gatsby's [webpack config](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/utils/webpack.config.js), which is also linked in the above doc. That's why I think it's an advanced use case and not too sure if it belongs to the doc.

@wardpeet  Would love to hear your opinion on this.
